src/NServiceBus.Metrics/RawData/OccurrenceWriter.cs

```diff
@@ -0,0 +1,42 @@
+namespace NServiceBus.Metrics.RawData
+{
+    using System;
+    using System.IO;
+
+    static class OccurrenceWriter
+    {
+        const long Version = 1;
+
+        // WIRE FORMAT, Version: 1
+
+        //0                   1                   2                   3
+        //0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
```

Comment:
My opinion is `yes`. It's based on the following assumptions:
- we use (smart) batching and ship data only if they are in there, batched up to 4k of entries
- format is much more dense than JSON and can be easily read and consumed on the receiver side
- we could have one reporter asking multiple rings for data, but then, we'd need to come up with even more fancy encoding and marking the message with data types that it includes.

I think that if we want to continue this discussion, we should bring some numbers.